frontend/src/components/AccountDetailsEditContainer.vue

```diff
@@ -0,0 +1,259 @@
+            <td class="p-1">
+              <TextInputComponent
+                :disabled="disableInputShowSpinner || deleted"
+                :hidden="true"
+                place-holder="Passwort wiederholen"
+                @input-changed="(val) => (pwFieldControl = val)"
+                :max-length="128"
+              />
+            </td>
+          </tr>
+          <tr :class="showadminview ? '' : 'collapse'">
+            <td></td>
+            <td>
+              <div>
+                <input
+                  :disabled="disableInputShowSpinner || deleted"
+                  type="checkbox"
+                  class="cursor-pointer m-1"
+                  v-model="isAdmin"
+                  :checked="isAdmin"
+                />
+                <label class="cursor-pointer" @click="toggleCheckBox">
+                  Adminaccount
+                </label>
+              </div>
+            </td>
+          </tr>
+        </table>
+        <div class="flex items-center pt-4">
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="reset()"
+          >
+            Abbrechen
+          </ButtonComponent>
+          <div class="flex-grow text-center">
+            <div class="text-red-500 italic" v-if="displayError">
+              Ungültige Eingabe
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="updateInformation()"
+          >
+            Speichern
+          </ButtonComponent>
+        </div>
+        <div :class="showadminview ? 'flex items-center pt-1' : 'collapse'">
+          <div class="flex-grow text-center">
+            <div class="text-green-500 italic" v-if="deleted">
+              Account gelöscht
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="deleteAccount()"
+          >
+            Löschen
+          </ButtonComponent>
+        </div>
+      </template>
+    </ContainerComponent>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+import TextInputComponent from './TextInputComponent.vue';
+import ButtonComponent from './ButtonComponent.vue';
+import ContainerComponent from './ContainerComponent.vue';
+import { IUser, UserClass } from '../../../model/user/IUser';
+import { AccountRestInterface } from '../restInterfaces/AccountRestInterface';
+import {
+  LoernwerkError,
+  LoernwerkErrorCodes,
+} from '../../../backend/loernwerkError';
+import { router } from '../router';
+
+const props = defineProps({
+  /**
+   * Whether the delete button is displayed
+   */
+  showadminview: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  userid: {
+    type: Number,
+    required: false,
+    default: null,
+  },
+});
+
+const nameField = ref('');
```

Comment:
We could pre-fill all of these if we set them after we query the account, correct?

Reply:
shouldn't matter, on an edit they are correctly filled. but if you are not changing something and they are empty (="") im not sending a request. But i guess it would be better to fill them and checking if they changed from the fetched user

Reply:
Not sure if checking is even required, even if we send a request though nothing changed, I dont think it's that bad